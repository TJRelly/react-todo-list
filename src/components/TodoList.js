import { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import "../static/TodoList.css";

const TodoList = ({ title }) => {
    //Renders list of Todo Form and list of Todos
    const [todos, setTodos] = useState([]);
    const addTodo = (newTodo) => setTodos((todos) => [...todos, newTodo]);
    const removeTodo = id => setTodos(todos => todos.filter(todo => todo.id !== id))

    return (
        <div className="TodoList d-flex">
            <NewTodoForm addTodo={addTodo} />
            <div className="TodoList-list">
                <h1>{title}</h1>
                <ul>
                    {todos.map((todo, i) => (
                        <li>
                            <Todo {...todo} key={i} remove={removeTodo}/>
                        </li>
                    ))}
                </ul>
            </div>
        </div>
    );
};

export default TodoList;
