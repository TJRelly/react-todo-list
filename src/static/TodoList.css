.TodoList {
    display: grid;
    grid-template-columns: 1fr 1fr;
}

.TodoList-list {
    padding: 1rem 2rem;
}

.TodoList ul {
    list-style: none;
    padding: 0;
    display: flex;
    flex-direction: column;
    align-items: start;
}

.TodoList li {
    text-align: start;
}

@media (max-width: 600px) {
    .TodoList {
        grid-template-columns: 1fr;
    }
}

@media (max-width: 600px) {
    .TodoList {
        display: flex;
        flex-direction: column;
    }
}
