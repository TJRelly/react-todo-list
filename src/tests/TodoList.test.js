import { render, fireEvent } from "@testing-library/react";
import TodoList from "../components/TodoList";

test("renders learn react link", () => {
    render(<TodoList />);
});

test("matches snapshot", function () {
    const { asFragment } = render(<TodoList title="Title" />);
    expect(asFragment()).toMatchSnapshot();
});

test("can add a new todo", function () {
    const todoList = render(<TodoList />);

    // no todos yet
    expect(todoList.queryByText("delete")).not.toBeInTheDocument();

    // Simulate filling out the form to add a new todo
    const textInput = todoList.getByLabelText("What do you need todo?");
    const addButton = todoList.queryByText("Add");

    // Change values in the form
    fireEvent.change(textInput, { target: { value: "do todos" } });

    // Submit the form to add the todo
    fireEvent.click(addButton);

    // expect to see a todo
    const removeButton = todoList.getByText("delete");
    expect(removeButton).toBeInTheDocument();
});

it("can remove a todo", function () {
    const todoList = render(<TodoList />);

    // Simulate filling out the form to add a new todo
    const textInput = todoList.getByLabelText("What do you need todo?");
    const addButton = todoList.getByText("Add");

    // Change values in the form
    fireEvent.change(textInput, { target: { value: "do todos" } });

    // Submit the form to add the todo
    fireEvent.click(addButton);

    const removeButton = todoList.getByText("delete");

    // click the remove button and the todo should be gone
    fireEvent.click(removeButton);
    expect(removeButton).not.toBeInTheDocument();
});
