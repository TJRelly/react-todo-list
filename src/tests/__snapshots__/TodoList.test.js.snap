// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList d-flex"
  >
    <form
      class="Todo-form"
    >
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="text"
        >
          What do you need todo?
        </label>
        <input
          class="form-control"
          id="text"
          name="text"
          placeholder="Enter a new todo"
          type="string"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Add
      </button>
    </form>
    <div
      class="TodoList-list"
    >
      <h1>
        Title
      </h1>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
